app-connector-client/server/resources/api.yaml

```diff
@@ -1,430 +1,590 @@
-                consoleUrl:
-                  type: 'string'
-                  description: 'URL on connected Kyma cluster to browse console'
-                domain:
-                  type: 'string'
-                  description: 'Domain of the connected Kyma cluster'
-                app:
-                  type: 'string'
-                  description: 'Application on connected Kyma cluster'
-       '401':
-          description: 'Connection is not authorized, usually the provided token is not valid'
+              $ref: "#/components/schemas/ConnectionDetails"
+      responses:
+        "200":
+          description: "Successful connection to Kyma"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ConnectionResponse"
+        "401":
+          description: "Connection is not authorized, usually the provided token is not valid"
           content:
-            text/plain:
+            application/json:
               schema:
-                type: 'string'
-                example: There is an error while registering. Please make sure that your token is unique and valid
+                type: "object"
+                properties:
+                  error:
+                    type: string
+                    example: There is an error while registering. Please make sure that your token is unique
     get:
+      tags:
+        - "connection"
       summary: Returns active connection info
       responses:
-        '200':
+        "200":
           description: "Currently connected application and cluster"
           content:
             application/json:
               schema:
-                type: 'object'
-                properties:
-                  metadataUrl:
-                    type: 'string'
-                    description: 'URL on connected Kyma cluster to manage APIs'
-                  eventsUrl:
-                    type: 'string'
-                    description: 'URL on connected Kyma cluster to send events'
-                  consoleUrl:
-                    type: 'string'
-                    description: 'URL on connected Kyma cluster to browse console'
-                  domain:
-                    type: 'string'
-                    description: 'Domain of the connected Kyma cluster'
-                  app:
-                    type: 'string'
-                    description: 'Application on connected Kyma cluster'
+                $ref: "#/components/schemas/ConnectionResponse"
+        "400":
+          description: "Not connected to Kyma cluster"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
     delete:
+      tags:
+        - "connection"
       summary: Disconnects the mock from current kyma cluster
       responses:
         204:
           description: Disconnection was successful
+        400:
+          description: "Error"

```

Comment:
Please provide description on when it usually happens (probably not needed at all, see my other comment on connector.js)